cmd/ssd/ssd.py

```diff
@@ -26,13 +34,13 @@ def check_iptables(name, plist):
 
     # get the ingress sandbox's docker_gwbridge network IP.
     # published ports get DNAT'ed to this IP.
-    ip = subprocess.check_output(['/usr/bin/nsenter', '--net=/var/run/docker/netns/ingress_sbox', '/bin/bash', '-c', 'ifconfig eth1 | grep \"inet\\ addr\" | cut -d: -f2 | cut -d\" \" -f1'])
+    ip = subprocess.check_output([ which("nsenter","/usr/bin/nsenter"), '--net=/var/run/docker/netns/ingress_sbox', which("bash", "/bin/bash"), '-c', 'ifconfig eth1 | grep \"inet\\ addr\" | cut -d: -f2 | cut -d\" \" -f1'])
     ip = ip.rstrip()
 
     for p in ports:
-        rule = '/sbin/iptables -t nat -C DOCKER-INGRESS -p tcp --dport {0} -j DNAT --to {1}:{2}'.format(p[1], ip, p[1])
+        rule = which("iptables", "/sbin/iptables") + '-t nat -C DOCKER-INGRESS -p tcp --dport {0} -j DNAT --to {1}:{2}'.format(p[1], ip, p[1])
```

Comment:
just noticed that this is inside a loop so for each ports it is going to be checked, maybe a better approach was to add all the which on top of the file and call them once and maybe error out if some of the tools are missing like `which__or_die`, but you can follow up in a separate patch for it

Reply:
Yes, makes sense.
I will push a seperate PR with the suggested changes.
Thx